src/workerd/api/streams/common.h

```diff
@@ -420,6 +420,10 @@ class WritableStreamController {
   // Indicates that no additional data will be written to the controller. All
   // existing pending writes should be allowed to complete.
 
+  virtual jsg::Promise<void> flush(jsg::Lock& js, bool markAsHandler = false) = 0;
+  // Requests all pending data to the written. The returned promise is resolved when all
+  // pending writes have completed.
```

Comment:
I'm not sure this "requests" as much as it waits for pending data to be written.